sympy/integrals/manualintegrate.py

```diff
@@ -1424,6 +1424,35 @@ def trig_product_rule(integral: IntegralInfo):
         return CscCotRule(integrand, symbol)
 
 
+def trig_cmplx_exp_rule(integral: IntegralInfo):
+    """
+    Strategy that rewrites sin, cos, sinh, and cosh in terms of complex exponentials.
+    Useful for integration techniques that handle exponentials better.
+    Applies only when the integrand belongs to a class that benefits from exponential rewriting,
+    such as combinations involving Error functions.
+
+    sin(x)  -> (exp(i*x) - exp(-i*x)) / (2*i)
+    cos(x)  -> (exp(i*x) + exp(-i*x)) / 2
+    sinh(x) -> (exp(x) - exp(-x)) / 2
+    cosh(x) -> (exp(x) + exp(-x)) / 2
+    """
+    integrand, symbol = integral
+    requires_exp_trig_rewriting = (erf, erfc, erfi)
+
+    if not any(integrand.has(f) for f in requires_exp_trig_rewriting):
+        return
+
+    # Replace trig and hyperbolic functions with their exponential forms
+    replaced = integrand.replace(
+        lambda expr: expr.func in (sin, cos, sinh, cosh),
+        lambda expr: expr.rewrite(exp)
```

Comment:
You can do this directly with `rewrite`:
```python
In [6]: (sin(x) + cos(x)).rewrite([sin, exp], tan)
Out[6]: 
               ⎛x⎞  
          2⋅tan⎜─⎟  
               ⎝2⎠  
cos(x) + ───────────
            2⎛x⎞    
         tan ⎜─⎟ + 1
             ⎝2⎠ 
```
It is probably better though to only do this for the expressions that contain the integration variable.